bson/src/main/org/bson/codecs/AtomicIntegerCodec.java

```diff
@@ -21,21 +21,47 @@
 
 import java.util.concurrent.atomic.AtomicInteger;
 
+import static org.bson.assertions.Assertions.isTrueArgument;
+import static org.bson.codecs.NumberCodecHelper.DEFAULT_DELTA;
+import static org.bson.codecs.NumberCodecHelper.decodeNumber;
+
 /**
  * Encodes and decodes {@code AtomicInteger} objects.
  *
  * @since 3.0
  */
 
 public class AtomicIntegerCodec implements Codec<AtomicInteger> {
+    private final double delta;
+
+    /**
+     * Construct a new instance
+     */
+    public AtomicIntegerCodec() {
+        this(DEFAULT_DELTA);
+    }
+
+    /**
+     * Construct a new instance
+     *
+     * @param delta the maximum delta between {@code expected} and {@code actual} for which both numbers are still
+     * considered equal. Required when converting Double values to {@code AtomicInteger} values. Defaults to {@code 0.00000000000001d}.
```

Comment:
Why would anyone ever want a delta other than 0.0? That just results in information loss.